app/models/financial_transaction_type.rb

```diff
@@ -0,0 +1,21 @@
+class FinancialTransactionType < ActiveRecord::Base
+  belongs_to :financial_transaction_class
+  has_many :financial_transactions, dependent: :restrict_with_exception
+
+  validates :name, presence: true
+  validates_uniqueness_of :name
+  validates :financial_transaction_class, presence: true
+
+  before_destroy :check_last_financial_transaction_type
```

Comment:
Maybe something like `restrict_deleting_last_financial_transaction_type` would be just a bit clearer?